### ISSUE TITLE
Incorrect Numbers from Number Keypad on Apple Extended Keyboard

### ISSUE THREAD

**Issue by michaelsaul** [PowerShell/PowerShell#2475]
Steps to reproduce
------------------
Using number pad on Apple Extended Keyboard to enter numbers, I receive strange characters. Numbers typed using numbers above the QWERTY keyboard work fine.

Expected behavior
-----------------
(when using number pad on extended keyboard)
```
>123456789=/*-+
```
Actual behavior
---------------
```
>OqOrOsOtOuOvOwOxOy=/*OmOl
```
Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value                                           
----                           -----                                           
PSVersion                      6.0.0-alpha                                     
PSEdition                      Core                                            
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                         
BuildVersion                   3.0.0.0                                         
GitCommitId                    v6.0.0-alpha.10                                 
CLRVersion                                                                     
WSManStackVersion              3.0                                             
PSRemotingProtocolVersion      2.3                                             
SerializationVersion           1.1.0.1  
```

**Comment by andschwa** [comment]
@michaelsaul please test against `[console]::ReadKey()` and if your issue repros, close this issue and file against .NET Core.

**Comment by michaelsaul** [comment]
I was able to reproduce using the command. All of the numbers show up as the letter O with alt, shift modifiers. Note that when I type a key into the terminal and hit enter, a letter is left at the prompt. In the example below, I typed the number 0 on the number pad, and was left with a q character at the prompt.

```
PS /Users/michaelsaul> [console]::ReadKey()                                    
O
KeyChar Key  Modifiers
------- ---  ---------
      O   O Alt, Shift


PS /Users/michaelsaul> q           
```

**Comment by andschwa** [comment]
@michaelsaul one thing you'll want to look into is verifying your `TERM` environment variable and actual terminal emulator settings are consistent; I've found this to be the most likely cause of `System.Console` problems (as it's just delegating mostly to `libtermcap`).

**Comment by michaelsaul** [comment]
@andschwa I'll ask, is there a preferred term type? I have been happy using `xterm-256colors` for years. Seeing how that is the default for MacOS, I would think it would have the best support out of the box. A quick cycle through all of the Terminal.app builtin terminal declarations revealed the issue for all except `ansi`.

**Comment by andschwa** [comment]
@michaelsaul oh man, I just remembered this is a Terminal(.app) thing. Can you try:

> In the Terminal application:
> Click on the "Terminal" menu
> Click "Preferences"
> Click on the "Profiles" tab.
> Click on the "Advanced" tab (right hand side of the window)
> Uncheck "Allow VT100 application keypad mode"

There are a few other settings I can't recall at the moment that you may want to toggle too (I have had problems with `alt` being interpreted correctly until I fiddled with something).

**Comment by michaelsaul** [comment]
@andschwa That seems to make it work. I did create a custom Terminal.app profile (in blue no less!), maybe we can include a custom Terminal Profile in the installer package?

You were correct about the .NET Core Console Class. A quick C# console application executed from the default Terminal profile exhibits the same behavior.

```
            Console.WriteLine("What is your name?");
            var name = Console.ReadLine();
            Console.WriteLine("Hello, {0}!", name);
```
```
$ dotnet bin/Debug/netcoreapp1.0/dotNetConsole.dll 
What is your name?
OpOqOrOsOtOuOvOwOxOy
Hello, OpOqOrOsOtOuOvOwOxOy!
```

If a user is going to install .NET core and develop in it, it might be something that they can figure out, but it's probably worth mentioning in a readme.

I created a PowerShell terminal with settings that work for me here: https://gist.github.com/michaelsaul/fc289abc59c9742086635c80f19b7457

Thanks so much!
